### PR TITLE
Add jump_url to messageables

### LINKED ISSUE #396: Add .jump_url attribute to Messageable
### Summary

Add the `.jump_url` property to the `Messageable` mixin class

### What is the feature request for?

The core library

### The Problem

You currently have to manually format a Discord URL to jump to a channel, thread, DM, etc. By adding a `.jump_url` to the `Messageable` mixin, this functionality would be added to those classes.

### The Ideal Solution

`Message`s have a `.jump_url` property that formats a URL string for the client to jump to the message directly. The same implementation that creates the URL could be added to `Messageable`s so that functionality could be provided to channels, DMs, threads, etc.

### The Current Solution

_No response_

### Additional Context

As discussed (briefly) in the [#general](https://discord.com/channels/881118111967883295/881118112492191796/936677692626059284) channel of the Nextcord Server.

### PR DESCRIPTION
**Pull request by acco4321**
## Summary

Add jump_url to TextChannel, GroupChannel and DMChannel messageables. Closes #396 

## Checklist
- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

(Pretty sure I updated the docs, new to working with docustrings)

### CODE REVIEW

File: nextcord/channel.py [hunk 1]
```diff
@@ -261,6 +261,15 @@ def last_message(self) -> Optional[Message]:
         """
         return self._state._get_message(self.last_message_id) if self.last_message_id else None
 
+    @property
+    def jump_url(self) -> str:
+        """:class:`str`: Returns a URL that allows the client to jump to the referenced messageable.
+
+        .. versionadded:: 2.0
+        """
+        guild_id = self.guild.id
```
Comment (MaskDuck) [inline nextcord/channel.py:270]: why:tm:

Comment (MaskDuck) [reply nextcord/channel.py:270]:
```suggestion
```

File: nextcord/channel.py [hunk 2]
```diff
@@ -261,6 +261,15 @@ def last_message(self) -> Optional[Message]:
...
         return self._state._get_message(self.last_message_id) if self.last_message_id else None
 
+    @property
+    def jump_url(self) -> str:
+        """:class:`str`: Returns a URL that allows the client to jump to the referenced messageable.
+
+        .. versionadded:: 2.0
+        """
+        guild_id = self.guild.id 
+        return f'https://discord.com/channels/{guild_id}/{self.id}'
```
Comment (MaskDuck) [inline nextcord/channel.py:271]:
```suggestion
        return f'https://discord.com/channels/{self.guild.id}/{self.id}'
```

**Review by MaskDuck — CHANGES_REQUESTED** [review]

**Comment by DenverCoder1** [comment]
Maybe these could go in the ABCs instead?

GuildChannel and PrivateChannel?

**Comment by acco4321** [comment]
Added to `PrivateChannel`, `VoiceChannel`, `StageChannel` and `Threads`.

File: nextcord/abc.py [hunk 3]
```diff
@@ -523,6 +533,14 @@ def permissions_synced(self) -> bool:
         category = self.guild.get_channel(self.category_id)
         return bool(category and category.overwrites == self.overwrites)
 
+    @property
+    def jump_url(self) -> str:
```
[the remark this answers is not in the document]
Comment (acco4321) [reply nextcord/abc.py:537]: I'm not opposed to it being in `CategoryChannel`, since it's valid. If others are happy, I'll leave it in there (didn't realise I put it in there in the first place, lol).

**Comment by acco4321** [comment]
Don't mind the latest commits, just fixing a mistake (made by force pushing).

`jump_url` is now implemented on `GuildChannel` and `PrivateChannel` abcs.

File: nextcord/channel.py [hunk 4]
```diff
@@ -1609,7 +1609,7 @@ def _sorting_bucket(self) -> int:
         return ChannelType.text.value
 
     @property
-    def type(self) -> ChannelType:
+    def type(self) -> ChannelType:#
```
Comment (ooliver1) [inline nextcord/channel.py:1612]: unrelated change?

File: nextcord/abc.py [hunk 5]
```diff
@@ -251,6 +260,7 @@ class GuildChannel:
     category_id: Optional[int]
     _state: ConnectionState
     _overwrites: List[_Overwrites]
+    jump_url: str
```
Comment (ooliver1) [inline nextcord/abc.py:263]: do these even work having an attribute and property of the same name?

File: nextcord/channel.py [hunk 6]
```diff
@@ -1779,6 +1779,7 @@ def created_at(self) -> datetime.datetime:
         """:class:`datetime.datetime`: Returns the direct message channel's creation time in UTC."""
         return utils.snowflake_time(self.id)
 
+
```
Comment (ooliver1) [inline nextcord/channel.py:1782]: unrelated and incorrect addition of line?

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Review by ooliver1 — APPROVED** [review]
tested: text, dms, categories, voice (weird discord behaviour)

File: nextcord/abc.py [hunk 7]
```diff
@@ -523,6 +531,14 @@ def permissions_synced(self) -> bool:
         category = self.guild.get_channel(self.category_id)
         return bool(category and category.overwrites == self.overwrites)
 
+    @property
+    def jump_url(self) -> str:
+        """:class:`str`: Returns a URL that allows the client to jump to the referenced messageable.
```
Comment (ooliver1) [inline nextcord/abc.py:536]:
```suggestion
        """:class:`str`: Returns a URL that allows the client to jump to this channel.
```
gonna just commit myself to not bother you

File: nextcord/threads.py [hunk 8]
```diff
@@ -312,6 +312,14 @@ def category_id(self) -> Optional[int]:
             raise ClientException('Parent channel not found')
         return parent.category_id
 
+    @property
+    def jump_url(self) -> str:
+        """:class:`str`: Returns a URL that allows the client to jump to the referenced messageable.
```
Comment (ooliver1) [inline nextcord/threads.py:317]:
```suggestion
        """:class:`str`: Returns a URL that allows the client to jump to this channel.
```
